### ISSUE TITLE
Invalid use of rpcHandler and subscribeTopic in WakuRelay?

### ISSUE THREAD

**Issue by kdeme** [status-im/nim-waku#114]
I was having a better look at the v2 code to understand the flow of messages (posting, subscribing, filters & notification) and I find the current implementation rather weird, and probably not the way it is intended to be used.

- [rpcHandler](https://github.com/status-im/nim-waku/blob/master/waku/protocol/v2/waku_relay.nim#L114): the `filter.notify` is called there. This is rather strange and I think this should be done in the subscribe handler. From what I understand, the `rpcHandler` is used on what I'll call "raw" incoming messages, meaning any kind of message. I noticed for example that subscriptions are handled there (PubSub part), but also signature checking, and other custom validators (FloodSub part). Also checks are done to see if we are subscribed to the topic it is published on, and messages are forwarded/broadcasted in there.
Calling the `filter.notify` there will cause problems. e.g. as it will ignore the validation and signature checks. And it will pass on message in topics not subscribed to. So this functionality should be in the subscribe handler imo.

- [subscribeTopic](https://github.com/status-im/nim-waku/blob/master/waku/protocol/v2/waku_relay.nim#L83): This too looks weird, afaik this call is used when a peer gets subscribed or unsubscribe to a topic. But we actually do our own `w.subscribe` in it. Not sure what the idea is there? The `subscribe` calls are just done directly in wakunode anyhow (https://github.com/status-im/nim-waku/blob/master/waku/node/v2/wakunode2.nim#L240).

So in short, I think both these calls are not needed in `WakuRelay`

**Comment by oskarth** [comment]
> So this functionality should be in the subscribe handler imo.

Agree, I think this has come up in PR review before.

> subscribeTopic: This too looks weird, afaik this call is used when a peer gets subscribed or unsubscribe to a topic

This was just a stub (hence NOOP log) and came before the three protocol split. It was meant to be be an abstraction for Waku Topics IIRC. I think it is safe to remove it.

**Comment by kdeme** [comment]
OK, I'll clean this up.

**Comment by kdeme** [comment]
Resolved in #128
